force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -148,14 +149,25 @@ export default class ServiceDeliveryRow extends LightningElement {
     }
 
     handleInputChange(event) {
-        if (event.target.fieldName === this.fields.contact.fieldApiName) {
-            if (event.detail.value && event.detail.value.length) {
-                this.selectedContact = event.detail.value[0];
-                this.handleGetServicesEngagements(event.detail.value[0]);
+        if (this.firstFieldSetElement === this.fields.contact.fieldApiName) {
```

Comment:
Does Contact have to be the first field in the fieldset for this to work?

Reply:
Yes.. That is correct. If the contact is the first field then everything that is part of the if statement is executed else it moves on to else if.